src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -813,7 +814,21 @@ protected void runSpooler() {
         Set<CompletableFuture<?>> publishRequests = ConcurrentHashMap.newKeySet();
         while (!Thread.currentThread().isInterrupted()) {
             try {
-                getConnection(false).connect().get();
+                try {
+                    getConnection(false).connect().get();
+                } catch (ExecutionException e) {
+                    if (Utils.getUltimateCause(e) instanceof InterruptedException) {
+                        logger.atWarn().log("Shutting down spooler task");
+                        Thread.currentThread().interrupt();
+                        break;
+                    } else {
+                        logger.atError().log("Error when publishing from spooler", e.getCause());
+                        // Do not retry connecting immediately, most likely it would fail. Backoff and retry later.
+                        // It fails when PKCS11 is not available, for example, so retrying that immediately
+                        // will just spam the logs.
+                        Thread.sleep(Duration.ofMinutes(2).toMillis() + RANDOM.nextInt(10_000));
+                    }
```

Comment:
2 minutes seems extremely conservative. Is that how long it takes to resolve with PKCS11?

Is it possible to do exponential back-off with jitter just so a transient issue doesn't cause a 2 minute delay?

Otherwise could using a smaller baseline duration - say 30s with 30s of jitter?

Reply:
PKCS can take a while, but your point is of course valid. I used 2 minutes because it is what we use everywhere else, when subscribing to jobs and shadow topics for example.

Reply:
ok - maybe worth a revisit as separate issue then